import necessary librabies and packages like numpy and pandas

In [1]:
import pandas as pd
import numpy as np

read the csv file

In [2]:
terror=pd.read_csv('terror.csv', encoding='latin1')

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3049: DtypeWarning: Columns (4,61,62,66,116,117,123) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


preview of first 10 rows of data

In [3]:
terror.head(10)

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       0     0        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   
5  197001010002   1970       1     1        NaN         0        NaN      217   
6  197001020001   1970       1     2        NaN         0        NaN      218   
7  197001020002   1970       1     2        NaN         0        NaN      217   
8  197001020003   1970       1     2        NaN         0        NaN      217   
9  197001030001   1970       1     3        NaN         0        NaN      217   

          country_txt  region  ...  \
0  Dominican Republic       2  ...   
1              Mexico       1  ...   
2         Philippines       5  ...   
3              Greece       8  ...   
4               Japan       4  ...   
5       United States       1  ...   
6             Uruguay       3  ...   
7       United States       1  ...   
8       United States       1  ...   
9       United States       1  ...   

                                            addnotes  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
5  The Cairo Chief of Police, William Petersen, r...   
6                                                NaN   
7  Damages were estimated to be between $20,000-$...   
8  The New Years Gang issue a communiqué to a loc...   
9  Karl Armstrong's girlfriend, Lynn Schultz, dro...   

                                              scite1  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
5  "Police Chief Quits," Washington Post, January...   
6                                                NaN   
7  Committee on Government Operations United Stat...   
8  Tom Bates, "Rads: The 1970 Bombing of the Army...   
9  Committee on Government Operations United Stat...   

                                              scite2  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
5  "Cairo Police Chief Quits; Decries Local 'Mili...   
6                                                NaN   
7  Christopher Hewitt, "Political Violence and Te...   
8  David Newman, Sandra Sutherland, and Jon Stewa...   
9  Tom Bates, "Rads: The 1970 Bombing of the Army...   

                                              scite3        dbsource  INT_LOG  \
0                                                NaN            PGIS        0   
1                                                NaN            PGIS        0   
2                                                NaN            PGIS       -9   
3                                                NaN            PGIS       -9   
4                                                NaN            PGIS       -9   
5  Christopher Hewitt, "Political Violence and Te...  Hewitt Project       -9   
6                                                NaN            PGIS        0   
7                                                NaN  Hewitt Project       -9   
8  The Wisconsin Cartographers' Guild, "Wisconsin...  Hewitt Project        0   
9  David Newman, Sandra Sutherland, 

1.How many attacks happened in India?

In [4]:
terror[terror['country_txt']=='India']['eventid'].unique().size

9940

***9940 attacks happend in India***

2.How many attacks in india with atleast 3 ppl dead

In [5]:
#kills is given in the column nkills and country name is given in country_txt
terror.query('country_txt=="India" & nkill<=3')['nkill'].size

8362

***8362 attacks were happend in India where atleast 3 people dead***

3.Extract the city and summary for attacks above

In [6]:
terror.query('country_txt=="India" & nkill<=3')[['city','summary']].dropna(axis=0).head()

city                                            summary
67453   New Delhi  01/09/1998: At least 44 people were injured, t...
67555  Coimbatore  02/14/1998: Thirty-six people were killed and ...
67556  Coimbatore  02/14/1998: Thirty-six people were killed and ...
67557  Coimbatore  02/14/1998: Thirty-six people were killed and ...
67558  Coimbatore  02/14/1998: Thirty-six people were killed and ...

***the above dataframe shows the city and summary of attacks happend in India where atleast 3 people dead***

4.In a single terror incident in India, find out top 5 cities by number killed

In [7]:
terror.query("country_txt == 'India'").sort_values(by = "nkill",ascending= False)[['city', 'nkill']].head()

city  nkill
81000              Mumbai  188.0
96598             Jhargam  115.0
54339              Bombay  115.0
95860  Dantewada district   82.0
56837            Banabari   70.0

***the top 5 cities by number killed in a single terror incident in India***

5.In a single terror incident in India, find out top 5 cities by number killed and wounded

In [8]:
terror['killnwound'] = terror['nkill']+terror['nwound']
terror.query('country_txt=="India"').groupby('city').agg({'killnwound':'sum'}).sort_values(by='killnwound',ascending=False).head(6)

killnwound
city                 
Srinagar       3001.0
Unknown        2341.0
New Delhi      2011.0
Mumbai         2011.0
Jammu          1111.0
Guwahati        812.0

***top 5 cities by number killed and wounded in a single terror incident in India***

6.How many attacks were successful that were suicide attacks?

In [9]:
terror.query('suicide==1 & success==1')['suicide'].size

4260

***4260 attacks were successfull suicidal attacks***

7.label all the incidents where the number killed was more than 5 as severe. 

In [10]:
severe=[]
for i in terror['nkill']:
    if i>=5:
        severe.append('severe')
    else:
        severe.append('nonsever')
terror['severe'] = severe
terror.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       0     0        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... scite2 scite3 dbsource  INT_LOG  INT_IDEO  \
0  Dominican Republic       2  ...    NaN    NaN     PGIS        0         0   
1              Mexico       1  ...    NaN    NaN     PGIS        0         1   
2         Philippines       5  ...    NaN    NaN     PGIS       -9        -9   
3              Greece       8  ...    NaN    NaN     PGIS       -9        -9   
4               Japan       4  ...    NaN    NaN     PGIS       -9        -9   

   INT_MISC  INT_ANY related killnwound    severe  
0         0        0     NaN        1.0  nonsever  
1         1        1     NaN        0.0  nonsever  
2         1        1     NaN        1.0  nonsever  
3         1        1     NaN        NaN  nonsever  
4         1        1     NaN        NaN  nonsever  

[5 rows x 139 columns]

***above output shows all the incidents and number killed more than 5 labelled as "severe" in the column severe***

8.write a function to label an incident that was both successful and suicidal

In [11]:
def labelincident(terror,eventid):
    dict1 =terror[terror['eventid']==eventid]
    print(dict1['success'],':',dict1['suicide'])
    if (int(dict1['success'])==1) & (int(dict1['suicide'])==1):
        return True
    else:
        return False
labelincident(terror,int(input("Enter an event id")))

Enter an event id197000000001
0    1
Name: success, dtype: int64 : 0    0
Name: suicide, dtype: int64


False

***above function returns the incidents which were both successful and suicidal by giving the eventid***

9.Create a new category representing if the incident occured in Afghanistan, Pakistan or India as one level of the category and all the other countries as another level

In [12]:
countries = list(terror['country_txt'])
category = []
countrytrue = ['India','Pakistan','Afghanistan']
for country in countries:
    if country in countrytrue:
        category.append('1')
    else:
        category.append('0')
terror['category'] = category
terror.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       0     0        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... scite3 dbsource INT_LOG  INT_IDEO  \
0  Dominican Republic       2  ...    NaN     PGIS       0         0   
1              Mexico       1  ...    NaN     PGIS       0         1   
2         Philippines       5  ...    NaN     PGIS      -9        -9   
3              Greece       8  ...    NaN     PGIS      -9        -9   
4               Japan       4  ...    NaN     PGIS      -9        -9   

   INT_MISC  INT_ANY  related killnwound    severe  category  
0         0        0      NaN        1.0  nonsever         0  
1         1        1      NaN        0.0  nonsever         0  
2         1        1      NaN        1.0  nonsever         0  
3         1        1      NaN        NaN  nonsever         0  
4         1        1      NaN        NaN  nonsever         0  

[5 rows x 140 columns]

***above represent a new category of countries
1 level:if the incident occured in Afghanistan, Pakistan or India 
0 level: for all the other countries***

10.How many incidents happened in Af-Pak-India vs ROW?

In [13]:
terror.query("(country_txt=='India')| (country_txt == 'Pakistan')| (country_txt == 'Afganistan')").shape[0]

22708

11.List the number of suicides attacks and average kills by Af-Pak-India vs ROW. Rename columns in the output as Average_Kills and Number_Incidents.

In [14]:
terror.groupby('category').agg({'suicide':'count','nkill':'mean'}).rename(columns={'suicide':'Number_Incident','nkill':'Average_Kills'})

Number_Incident  Average_Kills
category                                
0                  124374       2.428727
1                   32398       2.105574